### ISSUE TITLE
Ubiquiti Unifi Controller

### ISSUE THREAD

**Issue by summersab** [turnkeylinux/tracker#1312]
I think it would be interesting to have a Ubiquiti Unifi controller VM. For those not familiar, here's a briefer on how a Unifi controller works. Basically, on your normal router, all of the management software is on the hardware itself, and you have to go to 192.168.1.1 to manage the settings. For Ubiquiti, you install that software on a separate system. This can be done in a few ways:
- Install it locally on something like a RPi (the end result is the same setup as a standard router, but the management software is just separate)
- Purchase a $99USD Unifi Cloud Key that is basically the same as the previous option
- Install it on a VM in the cloud (I have a TKL image for mine). This makes it possible to manage the network from anywhere

Once you do this, the system can either run on its own or be connected to Unifi's cloud platform (this is the most common option). It's behind 2FA, way more secure, and allows administrators to centralize and manage multiple networks from a single portal.

The dependencies are very simple. Ubiquiti provides a *.deb package for their Unifi controller package. It has a few dependencies like Mongo DB. The one outlier is that it requires `java8-runtime-headless`, and I believe this requires using a backports repo. Other than that, there's not much to it.

Based on a quick search of the AMI library, there isn't an off-the-shelf Unifi controller image, so this might be a great way to get the attention of sysadmins and have them take a closer look at TKL.

**Comment by JedMeister** [comment]
Sounds awesome! :smile:

**Comment by Dude4Linux** [comment]
Interesting.  I created a Unifi Controller appliance as a one-off project for a church where I installed WiFi a number of years ago, so it's definitely doable.  It was running Jessie if I remember correctly.  I didn't bother pushing it to an official appliance because I didn't see the interest.  I now have both Unifi and UNMS controllers running in a Raspberry Pi 4 managing my home/business network.  If there is sufficient interest, I'd consider dusting this off and updating it for 16.0.  Only problem I can think of is that UNMS wants to run in a Docker container.  See https://github.com/Dude4Linux/unifi

**Comment by l-arnold** [comment]
Interested!  (nudge).  Also @Dude4Linux  is it relatively easy to put TKL Images to a Rasberry Pi?

**Comment by JedMeister** [comment]
> Interested! (nudge). Also @Dude4Linux is it relatively easy to put TKL Images to a Rasberry Pi?

Our current images are only for `amd64` (i.e. 64 bit x86). The Raspberry Pi have an ARM cpu so it's not at all compatible. Supporting Raspberry Pi is something we've always wanted, but it's not a priority, so I'm not at all sure when we might get to it.

**Comment by speedy32129** [comment]
Also interested (nudge).

**Comment by JedMeister** [comment]
Yeah, I think that this might be a great appliance to build. Unfortunately we have a lot on and not too many spare cycles. If you want to see this in action anytime soon, someone will need to take the lead on development.

The repo posted above seems like a pretty good starting point?!

A couple of notes:
- Due to licencing changes, MongoDB is no longer available from Debian repos (it's still open source, but now has limitations on it's use when provided as a service for third parties). They do have their own apt repo, so installing from upstream is an option. Note that newer MongoDB builds have a CPU requirement that not all hardware will have (I haven't actually confirmed what isn't compatible - but it segfaults on my home server).
- Java 8 is no longer available via Debian backports. However, there is an organisation related to Java development (can't remember it's name OTTOMH) that provide updated Java 8 builds (and they have an apt repo too).
- Last I checked, the software license is non-free (proprietary freeware). So providing an appliance for this would be a bit of a departure from how we've done things historically (all of our appliances to date have always only included FLOSS - free libre open source software). It also appears to explicitly have a clause noting that we are **NOT** permitted to:
> "distribute, [... or] transfer [...] Ubiquiti [software] or modifications thereof or accompanying documentation in any form to any person without the prior written consent of Ubiquiti.

That last point means that unlike existing appliances, the Ubiquiti software could not be pre-installed. It would need to be downloaded and installed on firstboot. That's probably doable, but it's not really how we like to do things...
